test/TrueLayer.Tests/Serialization/OneOfJsonConverterTests.cs

```diff
@@ -60,6 +61,27 @@ public void Can_read_from_status_discriminator()
             oneOf.Value.ShouldBeOfType<Bar>();
         }
 
+        [Fact]
+        public void Can_read_from_status_discriminator_Refund()
+        {
+            string json = @"{
+                    ""status"": ""failed"",
+                    ""AmountInMinor"": 1000,
+                    ""CreatedAt"": ""2021-01-01T00:00:00Z"",
+                    ""FailedAt"": ""2021-01-01T00:02:00Z"",
+                    ""FailureReason"": ""Something bad happened""
+                }
+            ";
+
+            var oneOf = JsonSerializer.Deserialize<OneOf<RefundPending, RefundAuthorized, RefundExecuted, RefundFailed>>(json, _options);
+            oneOf.Value.ShouldBeOfType<RefundFailed>();
+            oneOf.AsT3.AmountInMinor.ShouldBe<uint>(1000);
+            oneOf.AsT3.Status.ShouldBe("failed");
```

Comment:
The fact that we're inlining unions with `using` directives does not give us the option to define custom getters on the discriminated unions. I don't think users of the library should rely on positional notation for readability reasons, also we could break this very easily by adding a variant or changing the order of variants without noticing. Worth adding customised getters so that consumers of the library can easily discriminate ? Like `IsFailed()` or `asFailed()` which would hide the the `is/AsT_N_` notation. 
ps: this would be consistent with that we do on the Java lib as well

Reply:
I agree! That is another thing that is on my list of things to refactor.